lib/resolver.fallback.js

```diff
@@ -50,113 +53,108 @@ export default class PodletClientFallbackResolver {
-                ? this.#httpsAgent
-                : this.#httpAgent,
-                uri: outgoing.fallbackUri,
-                headers,
-            };
-
-            const timer = this.#histogram.timer({
+            // Response is OK. Store response body as fallback html for caching
+            timer({
                 labels: {
-                    podlet: outgoing.name,
+                    status: 'success',
                 },
             });
 
-            request(reqOptions, (error, res, body) => {
-                this.#log.debug(
-                    `start reading fallback content from remote resource - resource: ${outgoing.name} - url: ${outgoing.fallbackUri}`,
-                );
+            // Set fallback to the fetched content
+            outgoing.fallback = await body.text();
 
-                // Network error
-                if (error) {
-                    timer({
-                        labels: {
-                            status: 'failure',
-                        },
-                    });
-
-                    this.#log.warn(
-                        `could not create network connection to remote resource for fallback content - resource: ${outgoing.name} - url: ${outgoing.fallbackUri}`,
-                    );
-
-                    outgoing.fallback = '';
-                    resolve(outgoing);
-                    return;
-                }
-
-                // Remote responds but with an http error code
-                const resError = res.statusCode !== 200;
-                if (resError) {
-                    timer({
-                        labels: {
-                            status: 'failure',
-                        },
-                    });
-
-                    this.#log.warn(
-                        `remote resource responded with non 200 http status code for fallback content - code: ${res.statusCode} - resource: ${outgoing.name} - url: ${outgoing.fallbackUri}`,
-                    );
-
-                    outgoing.fallback = '';
-                    resolve(outgoing);
-                    return;
-                }
-
-                // Response is OK. Store response body as fallback html for caching
-                timer({
-                    labels: {
-                        status: 'success',
-                    },
-                });
+            this.#log.debug(
+                `successfully read fallback from remote resource - resource: ${outgoing.name} - url: ${outgoing.fallbackUri}`,
+            );
+            return outgoing;
+        } catch (error) {
+            timer({
+                labels: {
+                    status: 'failure',
+                },
+            });
 
-                // Set fallback to the fetched content
-                outgoing.fallback = body;
+            this.#log.warn(
+                `could not create network connection to remote resource for fallback content - resource: ${outgoing.name} - url: ${outgoing.fallbackUri}`,
```

Comment:
I think we are handling this case unless Undici errors are different from request... which might well be.

Before request was like this:

```js
// request error
r.on('error', error => {
  // handle network error
})

// undici error
try {}
catch (error) {
  // handle network error
}

// request non 200 codes
r.on('response', response => {
  const resError = response.statusCode !== 200
});

// undici non 200 codes
const { statusCode, body } = await request(...);
const resError = statusCode !== 200;
```

Or is it still missing something you think?

Reply:
Right. I think what I am missing is distinguished log lines for each scenario. Has that been lost here or is it just me which have gone blind.

Reply:
I didn't remove any error messages, things just got moved around but I can double check to make sure I haven't messed anything up.

Reply:
Taking a look at the content resolver, every success/error/log/warning etc seems to be have been mapped 1-1 from reject/resolve to throw/return

eg.
```js
// before
reject(badGateway(`Recursion detected - failed to resolve fetching of podlet ${outgoing.recursions} times`);

// after
throw badGateway(`Recursion detected - failed to resolve fetching of podlet ${outgoing.recursions} times`);
```

I did notice that I've messed up incorporating the PR you made recently to lower the warn to debug. Need to fix that but otherwise I think it's as it should be.

Reply:
Then we're good :+1: